File: src/main/java/ordilov/lissn/member/application/port/in/MemberCommand.java
package ordilov.lissn.member.application.port.in;

import lombok.Value;
import ordilov.lissn.member.domain.AuthProvider;
import ordilov.lissn.member.domain.Member;

public class MemberCommand {

  @Value
  public static class RegisterCommand {
    String id;
    String name;
    String email;
    String picture;
    AuthProvider provider;

    public Member toEntity() {
      return Member.builder()
          .name(name)
          .email(email)
          .provider(provider)
          .picture(picture)
          .build();
    }


  }

  @Value
  public static class UpdateCommand {
    Long id;
    String name;
    String picture;
  }


}

File: src/main/java/ordilov/lissn/member/application/port/in/MemberCommand.kt
package ordilov.lissn.member.application.port.`in`

import ordilov.lissn.member.domain.AuthProvider

data class RegisterCommand(
    val id: String,
    val name: String,
    val email: String,
    val picture: String,
    val provider: AuthProvider
) {
}

data class UpdateCommand(
    val id: Long,
    val name: String,
    val picture: String
) {
}